quadriga/src/test/java/edu/asu/diging/quadriga/core/service/impl/CollectionManagerImplTest.java

```diff
@@ -44,11 +48,215 @@ public boolean matches(Collection arg0) {
                 return (arg0.getName().equals(name) && arg0.getDescription().equals(desc));
             }
         }))).thenReturn(savedCollection);
-        
+
         Collection collection = managerToTest.addCollection(name, desc);
         Assert.assertEquals(id, collection.getId());
         Assert.assertEquals(name, collection.getName());
         Assert.assertEquals(desc, collection.getDescription());
     }
 
+    @Test
+    public void test_findCollection_success() {
+        Collection collection = new Collection();
+        ObjectId objectId = new ObjectId();
+        collection.setId(objectId);
+        Mockito.when(collectionRepo.findBy_id(objectId)).thenReturn(collection);
+        Collection foundCollection = managerToTest.findCollection(objectId.toString());
+        Assert.assertEquals(collection.getId().toString(), foundCollection.getId().toString());
+    }
+
+    @Test
+    public void test_editCollection_success() {
+
+        String name = CollectionLiterals.COLLECTION_NAME;
+        String desc = CollectionLiterals.COLLECTION_DESC;
+        Collection existingCollection = new Collection();
+        ObjectId id = new ObjectId();
+        existingCollection.setId(id);
+        existingCollection.setName(name);
+        existingCollection.setDescription(desc);
+
+        Mockito.when(managerToTest.findCollection(id.toString())).thenReturn(existingCollection);
```

Comment:
managerToTest is the class you're trying to test. Don't mock it.

Reply:
Oh alright, I should be mocking the repo, got it.
Will update this